app/models/speech.rb

```diff
@@ -52,7 +52,11 @@ def delivered_by_minister?
 private
 
   def date_for_government
-    delivered_on.try(:to_date)
+    if delivered_on && delivered_on.past?
+      delivered_on
```

Comment:
`delivered_on` is a time, in the `super` method `to_date` is always called, I assume so that a Government can be found at the start of the day the speech was delivered on. We should probably stick with that approach here.